pkg/command/service/load.go

```diff
@@ -0,0 +1,423 @@
+		return "", loadResult, fmt.Errorf("failed to get the cluster endpoint: %w", err)
+	}
+
+	// Prepare for load test
+	cmd, wrkLua, err := loadCmdBuilder(inputs, endpoint, namespace, svc)
+	if err != nil {
+		return "", loadResult, err
+	}
+
+	defer func() {
+		// Delete wrk lua script
+		if strings.EqualFold(inputs.LoadTool, "wrk") {
+			_, fileError := os.Stat(wrkLua)
+			if fileError == nil {
+				removeError := os.Remove(wrkLua)
+				if removeError != nil {
+					log.Printf("remove %s error : %s", wrkLua, removeError)
+				}
+			}
+		}
+	}()
+
+	loadStart := time.Now()
+	m := fmt.Sprintf("Namespace %s, Service %s, load start", namespace, svc.Name)
+	log.Println(m)
+
+	go func() {
+		runCmd := exec.Command("/bin/sh", "-c", cmd)
+		var output []byte
+		output, err = runCmd.Output()
+		loadOutput = string(output)
+
+		if err != nil {
+			m := fmt.Sprintf("run load command error: %s", err)
+			fmt.Println(m)
+			return
+		}
+
+		loadEnd := time.Now()
+		loadDuration := loadEnd.Sub(loadStart)
+		m := fmt.Sprintf("Namespace %s, Service %s, load end, take off %.3f seconds", namespace, svc.Name, loadDuration.Seconds())
+		log.Println(m)
+
+		time.Sleep(inputs.WaitPodsReadyDuration * time.Second) //wait for all pods ready
+		pdch <- struct{}{}
+	}()
+
+	var preReadyReplicas int
+	for {
+		select {
+		case event := <-rdch:
+			// get replicas ready duration by watching deployment event
+			var r pkg.LoadReplicaResult
+			dm := event.Object.(*v1.Deployment)
+			readyReplicas := int(dm.Status.ReadyReplicas)
+			if event.Type == watch.Modified && readyReplicas > preReadyReplicas {
+				r.ReplicaReadyTime = time.Now()
+				r.ReadyReplicasCount = readyReplicas
+				r.ReplicaReadyDuration = r.ReplicaReadyTime.Sub(loadStart).Seconds()
+				replicaResults = append(replicaResults, r)
+				preReadyReplicas = readyReplicas
+			}
+		case <-pdch:
+			// get pods ready duration by pod conditions
+			var r pkg.LoadPodResult
+			podList, err := getSvcPods(ctx, params, namespace, svc.Name)
+			if err != nil {
+				return "", loadResult, err
+			}
+			for i := 0; i < len(podList); i++ {
+				pod := podList[i]
+				podCreatedTime := pod.GetCreationTimestamp().Rfc3339Copy()
+				present, PodReadyCondition := getPodCondition(&pod.Status, corev1.PodReady)
+				if present == -1 {
+					log.Println("failed to find Pod Condition PodReady and skip measuring")
+					continue
+				}
+				podReadyTime := PodReadyCondition.LastTransitionTime.Rfc3339Copy()
+				podReadyDuration := podReadyTime.Sub(podCreatedTime.Time).Seconds()
+				r.PodCreateTime = podCreatedTime
+				r.PodReadyTime = podReadyTime
+				r.PodReadyDuration = podReadyDuration
+				podResults = append(podResults, r)
+			}
+			var totalReadyReplicas int
+			for _, value := range replicaResults {
+				if value.ReadyReplicasCount > totalReadyReplicas {
+					totalReadyReplicas = value.ReadyReplicasCount
+				}
+			}
+			loadResult.TotalReadyReplicas = totalReadyReplicas
+			loadResult.TotalReadyPods = len(podResults)
```

Comment:
I just wonder if we need to calculate `TotalReadyReplicas` separately. Is the total ready replicas same as the number of ready pods?